controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java

```diff
@@ -642,6 +644,52 @@ public void testUpdateStreamState() throws Exception {
         response.close();
     }
 
+    /**
+     * Test for getScalingEvents REST API.
+     */
+    @Test
+    public void testGetScalingEvents() throws Exception {
+        final String resourceURI = getURI() + "v1/scopes/scope1/streams/stream1/scaling-events";
+
+        long fromDateTime = System.currentTimeMillis();
+        // test to get scaling events
+        Segment segment1 = new Segment(0, System.currentTimeMillis(), 0.00, 0.50);
+        Segment segment2 = new Segment(1, System.currentTimeMillis(), 0.50, 1.00);
+        List<Segment> segmentList1 = Arrays.asList(segment1, segment2);
+
+        Segment segment3 = new Segment(0, System.currentTimeMillis(), 0.00, 0.25);
+        Segment segment4 = new Segment(1, System.currentTimeMillis(), 0.25, 1.00);
+        List<Segment> segmentList2 = Arrays.asList(segment3, segment4);
+
+        ScaleMetadata scaleMetadata1 = new ScaleMetadata(System.currentTimeMillis(), segmentList1);
+        ScaleMetadata scaleMetadata2 = new ScaleMetadata(System.currentTimeMillis(), segmentList2);
+        List<ScaleMetadata> scaleMetadataList = Arrays.asList(scaleMetadata1, scaleMetadata2);
+        long toDateTime = System.currentTimeMillis();
+
+        when(mockControllerService.getScaleRecords(scope1, stream1)).
+                thenReturn(CompletableFuture.completedFuture(scaleMetadataList));
+        Response response = client.target(resourceURI).queryParam("from", fromDateTime).
+                queryParam("to", toDateTime).request().buildGet().invoke();
+        assertEquals("Get Scaling Events response code", 200, response.getStatus());
+        final List<ScaleMetadata> scaleMetadataListResponse = response.readEntity(List.class);
+        assertEquals("List Size", 3, scaleMetadataListResponse.size());
+
+        // Test for getScalingEvents for invalid scope.
+        final CompletableFuture<List<ScaleMetadata>> completableFuture1 = new CompletableFuture<>();
+        completableFuture1.completeExceptionally(new DataNotFoundException(""));
+        when(mockControllerService.getScaleRecords("scope1", "stream1")).thenReturn(completableFuture1);
```

Comment:
I think we are assuming that data not found means invalid scope, yes? because otherwise, `scope1` seems like a valid one. What if the stream doesn't exist? Is there any other scenario in which I can get a data not found, which is not via invalid scope?